src/nupic/research/connections.py

```diff
@@ -251,50 +487,90 @@ def computeActivity(self, activeInput, activePermanenceThreshold,
                                                synapse to be considered matching
     @param matchingSynapseThreshold    (int)   number of synapses needed for a
                                                segment to be considered matching
+    @param recordIteration             (bool)  bool to determine if we should
+                                               update the lastUsedIteration on
+                                               active segments and the internal
+                                               iteration variable
 
     @return (tuple) Contains:
                       `activeSegments`         (list),
-                      `matchingSegments`        (list),
+                      `matchingSegments`       (list),
 
     Notes:
-      activeSegments and matchingSegments are sorted by the cell they are on.
+      activeSegments and matchingSegments are sorted by the cell they are on
+      and they are lists of SegmentOverlap objects.
     """
 
-    numActiveSynapsesForSegment = [0] * self._nextSegmentIdx
-    numMatchingSynapsesForSegment = [0] * self._nextSegmentIdx
+    numActiveSynapsesForSegment = [0] * self._nextFlatIdx
+    numMatchingSynapsesForSegment = [0] * self._nextFlatIdx
 
     for cell in activeInput:
-      for synapseData in self.synapsesForPresynapticCell(cell).values():
-        segment = synapseData.segment
+      for synapse in self.synapsesForPresynapticCell(cell):
```

Comment:
I see the interface to Connections is changing slightly, which will break the PhasesTemporalMemory. So I think we'll need to either copy the old Connections to nupic.research, or change PhasesTemporalMemory to work with the updated Connections interface.

I see `synapsesForPresynapticCell` previously returned a dict, though its docstring says it was a set. Obviously your approach is better.